hnn_core/tests/test_gui.py

```diff
@@ -660,3 +669,36 @@ def test_gui_add_tonic_input(setup_gui):
                                gui.connectivity_widgets)
 
     assert _single_simulation['net'].external_biases['tonic'] is not None
+
+
+def test_update_nested_dict():
```

Comment:
Took me awhile to understand the logic but looks good now that I understand it :smile: